### DISCUSSION TITLE
0.8 Beta - Node Red Automation Examples

### DISCUSSION THREAD

**Discussion by jamos77** [blakeblackshear/frigate#353] [Show and tell]
Some simple node red automation examples for use with the 0.8 Beta:-
After importing the flow please see comments inside the node for more info.

![image](https://user-images.githubusercontent.com/43904804/101805948-45672a00-3b0b-11eb-84c7-bf582241c7ad.png)

Flow:  _(triple left click to select all)_
```
[{"id":"e48a29f2.c7e91","type":"mqtt in","z":"d86c352c.9e83d","name":"","topic":"frigate/events","qos":"2","datatype":"auto","broker":"3f5c82ac.348636","x":210,"y":220,"wires":[["9df79166.6c443"]]},{"id":"7b9880e4.24b028","type":"api-call-service","z":"d86c352c.9e83d","name":"notify MyMobile","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"notify","service":"mobile_app_myphone","entityId":"","data":"{\"data\":{\"attachment\":{\"url\":\"https://your.public.hass.address.com/api/frigate/notifications/{{payload.after.id}}.jpg\",\"content-type\":\"jpeg\"}},\"message\":\"Check Cameras if they look suspicious\",\"title\":\"Person Detected on {{payload.after.camera}} camera\"}","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":680,"y":220,"wires":[[]],"info":"Change the following:\nService- to be your mobile in Home Assistant\n\nDATA- url in attachment to be your Homeassistant public url. \n \nInfo:\nFor Android users you may need to add the following after the url:-\n?format=android\n\nI.E.\nhttps://your.public.hass.address.com/api/frigate/notifications/{{payload.after.id}}.jpg?format=android\n\n\n\n{{payload.after.camera}} = Name of Camera\n"},{"id":"9df79166.6c443","type":"json","z":"d86c352c.9e83d","name":"","property":"payload","action":"","pretty":false,"x":370,"y":220,"wires":[["b4265be9.a1b578"]],"info":"Convert JSON Strings to Objects"},{"id":"b4265be9.a1b578","type":"rbe","z":"d86c352c.9e83d","name":"","func":"rbe","gap":"","start":"","inout":"out","property":"payload.after.id","x":510,"y":220,"wires":[["7b9880e4.24b028"]],"info":"Block any Messages that have  duplicate After ID.  Cuts down on multiple mqtt messages for the same notification"},{"id":"75e669ab.c822f","type":"switch","z":"d86c352c.9e83d","name":"1.Front -- 2.Back cameras","property":"payload.after.camera","propertyType":"msg","rules":[{"t":"eq","v":"front","vt":"str"},{"t":"eq","v":"back","vt":"str"}],"checkall":"true","repair":false,"outputs":2,"x":710,"y":340,"wires":[["bfbc5e17.ba60e8"],["3fe075c1.e8243a"]],"info":"Change **front** and **back** Switch names to be the names of your cameras defined in the Frigate Config"},{"id":"bfbc5e17.ba60e8","type":"api-call-service","z":"d86c352c.9e83d","name":"Turn on Porch Lights","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"light","service":"turn_on","entityId":"light.porch_lights","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":940,"y":320,"wires":[[]]},{"id":"3fe075c1.e8243a","type":"api-call-service","z":"d86c352c.9e83d","name":"Turn on Garage Lights","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"light","service":"turn_on","entityId":"light.garage_led","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":940,"y":360,"wires":[[]]},{"id":"51be24a6.f68c9c","type":"mqtt in","z":"d86c352c.9e83d","name":"","topic":"frigate/events","qos":"2","datatype":"auto","broker":"3f5c82ac.348636","x":210,"y":340,"wires":[["ecebcde4.262678"]]},{"id":"ecebcde4.262678","type":"json","z":"d86c352c.9e83d","name":"","property":"payload","action":"","pretty":false,"x":370,"y":340,"wires":[["4c45fa7e.21cc64"]],"info":"Convert JSON Strings to Objects"},{"id":"4c45fa7e.21cc64","type":"rbe","z":"d86c352c.9e83d","name":"","func":"rbe","gap":"","start":"","inout":"out","property":"payload.after.id","x":510,"y":340,"wires":[["75e669ab.c822f"]],"info":"Block any Messages that have  duplicate After ID.  Cuts down on multiple mqtt messages for the same notification"},{"id":"a3ed0f30.68f8e","type":"mqtt in","z":"d86c352c.9e83d","name":"","topic":"frigate/events","qos":"2","datatype":"auto","broker":"3f5c82ac.348636","x":210,"y":460,"wires":[["e7f9181a.ea88f8"]]},{"id":"e7f9181a.ea88f8","type":"json","z":"d86c352c.9e83d","name":"","property":"payload","action":"","pretty":false,"x":370,"y":460,"wires":[["bdc5529d.c825d"]],"info":"Convert JSON Strings to Objects"},{"id":"bdc5529d.c825d","type":"rbe","z":"d86c352c.9e83d","name":"","func":"rbe","gap":"","start":"","inout":"out","property":"payload.after.id","x":510,"y":460,"wires":[["dee8ee79.d33d4"]],"info":"Block any Messages that have  duplicate After ID.  Cuts down on multiple mqtt messages for the same notification"},{"id":"dee8ee79.d33d4","type":"switch","z":"d86c352c.9e83d","name":"1.Person -- 2.Cat - Labels","property":"payload.after.label","propertyType":"msg","rules":[{"t":"eq","v":"person","vt":"str"},{"t":"eq","v":"cat","vt":"str"}],"checkall":"true","repair":false,"outputs":2,"x":690,"y":460,"wires":[["3d75b353.b4007c"],["218416fc.4b6062"]],"info":"Change **front** and **back** Switch names to be the names of your cameras defined in the Frigate Config"},{"id":"3d75b353.b4007c","type":"api-call-service","z":"d86c352c.9e83d","name":"Turn on Porch Lights","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"light","service":"turn_on","entityId":"light.porch_lights","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":940,"y":440,"wires":[[]]},{"id":"218416fc.4b6062","type":"api-call-service","z":"d86c352c.9e83d","name":"notify MyMobile","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"notify","service":"mobile_app_myphone","entityId":"","data":"{\"data\":{\"attachment\":{\"url\":\"https://your.public.hass.address.com/api/frigate/notifications/{{payload.after.id}}.jpg\",\"content-type\":\"jpeg\"}},\"message\":\"Check Cameras if they look suspicious\",\"title\":\"Person Detected on {{payload.after.camera}} camera\"}","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":920,"y":500,"wires":[[]],"info":"Change the following:\nService- to be your mobile in Home Assistant\n\nDATA- url in attachment to be your Homeassistant public url. \n \nInfo:\n{{payload.after.camera}} = Name of Camera\n"},{"id":"5e4df689.517bb","type":"comment","z":"d86c352c.9e83d","name":"Notify Phone if Frigate Detects Something","info":"Flow to notify phone if Frigate detects something","x":300,"y":160,"wires":[]},{"id":"57e99f0.2a2dbe","type":"comment","z":"d86c352c.9e83d","name":"Automation Based on Frigate Camera Name","info":"Flow to action something based on the camera name that triggered detction","x":310,"y":300,"wires":[]},{"id":"1c24dc24.431ca4","type":"comment","z":"d86c352c.9e83d","name":"Automation Based on Frigate Label Name","info":"Flow to turn on light if person detected or just notify phone if Frigate detects a cat","x":300,"y":420,"wires":[]},{"id":"7111e437.ec6adc","type":"mqtt in","z":"d86c352c.9e83d","name":"","topic":"frigate/study/person","qos":"2","datatype":"auto","broker":"3f5c82ac.348636","x":230,"y":620,"wires":[["e608767f.ba8178"]]},{"id":"4e0afa20.37f074","type":"switch","z":"d86c352c.9e83d","name":"1.person > 0 -- 2. person=0","property":"payload","propertyType":"msg","rules":[{"t":"gt","v":"0","vt":"num"},{"t":"lte","v":"0","vt":"num"}],"checkall":"true","repair":false,"outputs":2,"x":680,"y":620,"wires":[["9b799233.112298"],["74223f2e.0b178"]]},{"id":"e608767f.ba8178","type":"function","z":"d86c352c.9e83d","name":"To Number","func":"msg.payload = Number(msg.payload);\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":450,"y":620,"wires":[["4e0afa20.37f074"]],"info":"Convert msg.payload String to Number"},{"id":"9b799233.112298","type":"api-call-service","z":"d86c352c.9e83d","name":"","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"input_boolean","service":"turn_on","entityId":"input_boolean.study_person_detected","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":940,"y":580,"wires":[[]],"info":"Change the Input Boolean to one you have created in home assistant.\n\nTo create an Input Boolean in Home assistant:\nIn the UI:-\nConfiguration-->helpers-->Add Helper-->Toggle\n"},{"id":"74223f2e.0b178","type":"api-call-service","z":"d86c352c.9e83d","name":"","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"input_boolean","service":"turn_off","entityId":"input_boolean.study_person_detected","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":940,"y":640,"wires":[[]],"info":"Change the Entity ID Input Boolean to one you have created in home assistant.\n\nTo create an Input Boolean in Home assistant:\nIn the UI:-\nConfiguration-->helpers-->Add Helper-->Toggle"},{"id":"b3ddc950.0dde6","type":"comment","z":"d86c352c.9e83d","name":"Set an input boolean based on Label Count","info":"Flow to turn on light if person detected or just notify phone if Frigate detects a cat","x":300,"y":580,"wires":[]},{"id":"3f5c82ac.348636","type":"mqtt-broker","name":"","broker":"localhost","port":"1883","clientid":"","usetls":false,"compatmode":true,"keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","closeTopic":"","closeQos":"0","closePayload":"","willTopic":"","willQos":"0","willPayload":""},{"id":"57ce5691.fd6e48","type":"server","name":"Home Assistant","addon":true}]
```

**Comment by paularmstrong** [comment]
I do some similar actions. If there's a person out front, I first ring my "doorbell" (actually a more pleasant sound) to a speaker group in the house, then cast the HA view on a chromecast hub in my kitchen. 

If the sun is down and my xmas house lights and my standard house lights are off, then I turn the lights on while there are people, then turn them off again after 30 seconds.

Lastly, just for the front of house camera, I send myself a photo in a notification with information about exactly what zone the person is in.

<img width="1889" alt="Screen Shot 2020-12-10 at 12 00 06 PM" src="https://user-images.githubusercontent.com/33297/101823227-50a46080-3adf-11eb-8b1d-12cfd268476c.png">

**Reply by jamos77** [reply]
Hi Paul,  Any chance you can post the flow config for your Frigate/Events-Parse Function.  I'm always looking to improve my skills of using node red especially when it come to writing functions.

**Reply by paularmstrong** [reply]
Sure thing!

This is the parse function contents. I specifically try to pull out the label and zone so that I can fill the push notification message appropriately:

```js
msg.payload = JSON.parse(msg.payload);

const { camera, id, label, score, start_time, entered_zones } = msg.payload.after;
const startTime = new Date(parseInt(start_time, 10));

const zone = (entered_zones[0] || '').replace(/_/, ' ') || camera;

if (['person'].includes(label) && score >= 0.6) {
    return [{ payload: { id, label, score, startTime, zone } }, null];
}

return [null, msg];
```

And the full node-red flow (NOTE: You need to fix the domain on the push notification image URL and update the MQTT and HA connections)

```
[{"id":"fc28b771.75df48","type":"mqtt in","z":"46b4aa8a.842e34","name":"","topic":"frigate/events","qos":"2","datatype":"auto","broker":"b79e96fa.9c1e78","x":110,"y":760,"wires":[["af8a2ebc.21b1f"]]},{"id":"6e843857.37a688","type":"function","z":"46b4aa8a.842e34","name":"parse","func":"msg.payload = JSON.parse(msg.payload);\n\nconst { camera, id, label, score, start_time, entered_zones } = msg.payload.after;\nconst startTime = new Date(parseInt(start_time, 10));\n\nconst zone = (entered_zones[0] || '').replace(/_/, ' ') || camera;\n\nif (['person'].includes(label) && score >= 0.6) {\n    return [{ payload: { id, label, score, startTime, zone } }, null];\n}\n\nreturn [null, msg];","outputs":2,"noerr":0,"initialize":"","finalize":"","x":590,"y":760,"wires":[["7bad151c.14578c"],[]]},{"id":"6bfe5b3c.046404","type":"api-call-service","z":"46b4aa8a.842e34","name":"Send push","server":"74d5538b.e0092c","version":1,"debugenabled":false,"service_domain":"notify","service":"mobile_app_paul_pixel","entityId":"","data":"{\"message\":\"There's a {{payload.label}} at the {{payload.zone}}\",\"data\":{\"actions\": [{\"action\": \"UNLOCK\",\"title\":\"Unlock\"},{\"action\": \"SILENCE\", \"title\": \"Silence\"}], \"image\": \"https://MY-HA-DOMAIN/api/frigate/notifications/{{payload.id}}.jpg?format=android\" } }","dataType":"json","mergecontext":"","output_location":"payload","output_location_type":"msg","mustacheAltTags":false,"x":1010,"y":760,"wires":[[]]},{"id":"ab520f37.c71e4","type":"comment","z":"46b4aa8a.842e34","name":"Push notification","info":"","x":120,"y":720,"wires":[]},{"id":"af8a2ebc.21b1f","type":"api-current-state","z":"46b4aa8a.842e34","name":"Notifications enabled","server":"74d5538b.e0092c","version":1,"outputs":2,"halt_if":"on","halt_if_type":"str","halt_if_compare":"is","override_topic":false,"entity_id":"input_boolean.notifications","state_type":"str","state_location":"","override_payload":"none","entity_location":"","override_data":"none","blockInputOverrides":false,"x":360,"y":760,"wires":[["6e843857.37a688"],[]]},{"id":"7bad151c.14578c","type":"throttle","z":"46b4aa8a.842e34","name":"","throttleType":"time","timeLimit":"30","timeLimitType":"seconds","countLimit":0,"blockSize":0,"locked":false,"x":800,"y":760,"wires":[["6bfe5b3c.046404","e6da5648.f0ef58"]]},{"id":"b79e96fa.9c1e78","type":"mqtt-broker","name":"MQTT","broker":"10.0.1.210","port":"1883","clientid":"","usetls":false,"compatmode":false,"keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","closeTopic":"","closeQos":"0","closePayload":"","willTopic":"","willQos":"0","willPayload":""},{"id":"74d5538b.e0092c","type":"server","name":"Home Assistant","legacy":false,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":true}]
```

**Reply by jamos77** [reply]
Thanks, that's really useful for me.

**Comment by bm55** [comment]
> Some simple node red automation examples for use with the 0.8 Beta:-
> After importing the flow please see comments inside the node for more info.
> 
> ![image](https://user-images.githubusercontent.com/43904804/101805948-45672a00-3b0b-11eb-84c7-bf582241c7ad.png)
> 
> Flow: _(triple left click to select all)_
> 
> ```
> [{"id":"e48a29f2.c7e91","type":"mqtt in","z":"d86c352c.9e83d","name":"","topic":"frigate/events","qos":"2","datatype":"auto","broker":"3f5c82ac.348636","x":210,"y":220,"wires":[["9df79166.6c443"]]},{"id":"7b9880e4.24b028","type":"api-call-service","z":"d86c352c.9e83d","name":"notify MyMobile","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"notify","service":"mobile_app_myphone","entityId":"","data":"{\"data\":{\"attachment\":{\"url\":\"https://your.public.hass.address.com/api/frigate/notifications/{{payload.after.id}}.jpg\",\"content-type\":\"jpeg\"}},\"message\":\"Check Cameras if they look suspicious\",\"title\":\"Person Detected on {{payload.after.camera}} camera\"}","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":680,"y":220,"wires":[[]],"info":"Change the following:\nService- to be your mobile in Home Assistant\n\nDATA- url in attachment to be your Homeassistant public url. \n \nInfo:\nFor Android users you may need to add the following after the url:-\n?format=android\n\nI.E.\nhttps://your.public.hass.address.com/api/frigate/notifications/{{payload.after.id}}.jpg?format=android\n\n\n\n{{payload.after.camera}} = Name of Camera\n"},{"id":"9df79166.6c443","type":"json","z":"d86c352c.9e83d","name":"","property":"payload","action":"","pretty":false,"x":370,"y":220,"wires":[["b4265be9.a1b578"]],"info":"Convert JSON Strings to Objects"},{"id":"b4265be9.a1b578","type":"rbe","z":"d86c352c.9e83d","name":"","func":"rbe","gap":"","start":"","inout":"out","property":"payload.after.id","x":510,"y":220,"wires":[["7b9880e4.24b028"]],"info":"Block any Messages that have  duplicate After ID.  Cuts down on multiple mqtt messages for the same notification"},{"id":"75e669ab.c822f","type":"switch","z":"d86c352c.9e83d","name":"1.Front -- 2.Back cameras","property":"payload.after.camera","propertyType":"msg","rules":[{"t":"eq","v":"front","vt":"str"},{"t":"eq","v":"back","vt":"str"}],"checkall":"true","repair":false,"outputs":2,"x":710,"y":340,"wires":[["bfbc5e17.ba60e8"],["3fe075c1.e8243a"]],"info":"Change **front** and **back** Switch names to be the names of your cameras defined in the Frigate Config"},{"id":"bfbc5e17.ba60e8","type":"api-call-service","z":"d86c352c.9e83d","name":"Turn on Porch Lights","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"light","service":"turn_on","entityId":"light.porch_lights","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":940,"y":320,"wires":[[]]},{"id":"3fe075c1.e8243a","type":"api-call-service","z":"d86c352c.9e83d","name":"Turn on Garage Lights","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"light","service":"turn_on","entityId":"light.garage_led","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":940,"y":360,"wires":[[]]},{"id":"51be24a6.f68c9c","type":"mqtt in","z":"d86c352c.9e83d","name":"","topic":"frigate/events","qos":"2","datatype":"auto","broker":"3f5c82ac.348636","x":210,"y":340,"wires":[["ecebcde4.262678"]]},{"id":"ecebcde4.262678","type":"json","z":"d86c352c.9e83d","name":"","property":"payload","action":"","pretty":false,"x":370,"y":340,"wires":[["4c45fa7e.21cc64"]],"info":"Convert JSON Strings to Objects"},{"id":"4c45fa7e.21cc64","type":"rbe","z":"d86c352c.9e83d","name":"","func":"rbe","gap":"","start":"","inout":"out","property":"payload.after.id","x":510,"y":340,"wires":[["75e669ab.c822f"]],"info":"Block any Messages that have  duplicate After ID.  Cuts down on multiple mqtt messages for the same notification"},{"id":"a3ed0f30.68f8e","type":"mqtt in","z":"d86c352c.9e83d","name":"","topic":"frigate/events","qos":"2","datatype":"auto","broker":"3f5c82ac.348636","x":210,"y":460,"wires":[["e7f9181a.ea88f8"]]},{"id":"e7f9181a.ea88f8","type":"json","z":"d86c352c.9e83d","name":"","property":"payload","action":"","pretty":false,"x":370,"y":460,"wires":[["bdc5529d.c825d"]],"info":"Convert JSON Strings to Objects"},{"id":"bdc5529d.c825d","type":"rbe","z":"d86c352c.9e83d","name":"","func":"rbe","gap":"","start":"","inout":"out","property":"payload.after.id","x":510,"y":460,"wires":[["dee8ee79.d33d4"]],"info":"Block any Messages that have  duplicate After ID.  Cuts down on multiple mqtt messages for the same notification"},{"id":"dee8ee79.d33d4","type":"switch","z":"d86c352c.9e83d","name":"1.Person -- 2.Cat - Labels","property":"payload.after.label","propertyType":"msg","rules":[{"t":"eq","v":"person","vt":"str"},{"t":"eq","v":"cat","vt":"str"}],"checkall":"true","repair":false,"outputs":2,"x":690,"y":460,"wires":[["3d75b353.b4007c"],["218416fc.4b6062"]],"info":"Change **front** and **back** Switch names to be the names of your cameras defined in the Frigate Config"},{"id":"3d75b353.b4007c","type":"api-call-service","z":"d86c352c.9e83d","name":"Turn on Porch Lights","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"light","service":"turn_on","entityId":"light.porch_lights","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":940,"y":440,"wires":[[]]},{"id":"218416fc.4b6062","type":"api-call-service","z":"d86c352c.9e83d","name":"notify MyMobile","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"notify","service":"mobile_app_myphone","entityId":"","data":"{\"data\":{\"attachment\":{\"url\":\"https://your.public.hass.address.com/api/frigate/notifications/{{payload.after.id}}.jpg\",\"content-type\":\"jpeg\"}},\"message\":\"Check Cameras if they look suspicious\",\"title\":\"Person Detected on {{payload.after.camera}} camera\"}","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":920,"y":500,"wires":[[]],"info":"Change the following:\nService- to be your mobile in Home Assistant\n\nDATA- url in attachment to be your Homeassistant public url. \n \nInfo:\n{{payload.after.camera}} = Name of Camera\n"},{"id":"5e4df689.517bb","type":"comment","z":"d86c352c.9e83d","name":"Notify Phone if Frigate Detects Something","info":"Flow to notify phone if Frigate detects something","x":300,"y":160,"wires":[]},{"id":"57e99f0.2a2dbe","type":"comment","z":"d86c352c.9e83d","name":"Automation Based on Frigate Camera Name","info":"Flow to action something based on the camera name that triggered detction","x":310,"y":300,"wires":[]},{"id":"1c24dc24.431ca4","type":"comment","z":"d86c352c.9e83d","name":"Automation Based on Frigate Label Name","info":"Flow to turn on light if person detected or just notify phone if Frigate detects a cat","x":300,"y":420,"wires":[]},{"id":"7111e437.ec6adc","type":"mqtt in","z":"d86c352c.9e83d","name":"","topic":"frigate/study/person","qos":"2","datatype":"auto","broker":"3f5c82ac.348636","x":230,"y":620,"wires":[["e608767f.ba8178"]]},{"id":"4e0afa20.37f074","type":"switch","z":"d86c352c.9e83d","name":"1.person > 0 -- 2. person=0","property":"payload","propertyType":"msg","rules":[{"t":"gt","v":"0","vt":"num"},{"t":"lte","v":"0","vt":"num"}],"checkall":"true","repair":false,"outputs":2,"x":680,"y":620,"wires":[["9b799233.112298"],["74223f2e.0b178"]]},{"id":"e608767f.ba8178","type":"function","z":"d86c352c.9e83d","name":"To Number","func":"msg.payload = Number(msg.payload);\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":450,"y":620,"wires":[["4e0afa20.37f074"]],"info":"Convert msg.payload String to Number"},{"id":"9b799233.112298","type":"api-call-service","z":"d86c352c.9e83d","name":"","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"input_boolean","service":"turn_on","entityId":"input_boolean.study_person_detected","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":940,"y":580,"wires":[[]],"info":"Change the Input Boolean to one you have created in home assistant.\n\nTo create an Input Boolean in Home assistant:\nIn the UI:-\nConfiguration-->helpers-->Add Helper-->Toggle\n"},{"id":"74223f2e.0b178","type":"api-call-service","z":"d86c352c.9e83d","name":"","server":"57ce5691.fd6e48","version":1,"debugenabled":false,"service_domain":"input_boolean","service":"turn_off","entityId":"input_boolean.study_person_detected","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":940,"y":640,"wires":[[]],"info":"Change the Entity ID Input Boolean to one you have created in home assistant.\n\nTo create an Input Boolean in Home assistant:\nIn the UI:-\nConfiguration-->helpers-->Add Helper-->Toggle"},{"id":"b3ddc950.0dde6","type":"comment","z":"d86c352c.9e83d","name":"Set an input boolean based on Label Count","info":"Flow to turn on light if person detected or just notify phone if Frigate detects a cat","x":300,"y":580,"wires":[]},{"id":"3f5c82ac.348636","type":"mqtt-broker","name":"","broker":"localhost","port":"1883","clientid":"","usetls":false,"compatmode":true,"keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","closeTopic":"","closeQos":"0","closePayload":"","willTopic":"","willQos":"0","willPayload":""},{"id":"57ce5691.fd6e48","type":"server","name":"Home Assistant","addon":true}]
> ```

I copied this examples and sometimes I get several alerts because the ID changes within few seconds even though person is clearly in the view and tracking 
![image](https://user-images.githubusercontent.com/7528001/102132946-2e755000-3e09-11eb-9ae1-bff1b2a66b32.png)

**Reply by jamos77** [reply]
Yes I noticed that when I created them,  I got even more alerts without the the RBE node exception. You could use a delay node(limited to 1 message per number of seconds) in addition to or instead of the rbe node.  I think the multiple messages may be when it is getting a better percentage detection.

**Reply by blakeblackshear** [reply]
The `id` should stay the same for a single object. If your frame rate is too high on the camera, it can cause frigate to mark the object as "gone" too quickly and it will get a new `id`. I think it is 10 frames. So for a 10 fps camera, losing sight of the object for 1 second would cause it to get a new id. Also, if your min_score is too low, you can filter out a bunch of hits that would otherwise allow frigate to continue tracking the object even with a lower score.

**Reply by jamos77** [reply]
I also only alert when our HA Status is set to away, or the Alarm is armed, or if it is between midnight and 7 am so if someone was loitering outside our house I wouldn't mind the multiple alerts anyway.

**Reply by jamos77** [reply]
@blakeblackshear - one thing i've noticed is that if I stay still for a period of time and then move I will get a new id.  Does frigate ignore the detection of stationary objects after a period time?

**Reply by blakeblackshear** [reply]
That shouldn't happen unless it loses you. Stationary objects keep tracking.

**Reply by hawkeye217** [reply]
> The `id` should stay the same for a single object. If your frame rate is too high on the camera, it can cause frigate to mark the object as "gone" too quickly and it will get a new `id`. I think it is 10 frames. So for a 10 fps camera, losing sight of the object for 1 second would cause it to get a new id. 

Can you make that "10 frames" value configurable in a future version? Or just base it on the frame rate? I've run into that issue several times and I can't change the fps on one of my cheap cameras :(

**Reply by blakeblackshear** [reply]
Literally working on that today. You can also set the fps value in the camera config in frigate to lower it.

**Reply by hawkeye217** [reply]
Awesome!

**Reply by RealKanashii** [reply]
As far as the frigate is used as alert you really only need 1 alert per second o few seconds. You can put a delay node with "rate limit" to 2 seconds. It's also a nice trick to sync sources from different origins...
![rate limit](https://user-images.githubusercontent.com/9731708/102260392-aaa68b00-3f10-11eb-9183-2b2a400fec24.PNG)
This example block all measurements until it reach 10 minutes then pass it to the graph. That way you sync the data from different measurements. 

`[{"id":"11ce0725.dc9f09","type":"deconz-input","z":"335d5ee8.224672","name":"Sensor Temp Despatx","server":"ea12a4.b85e1d6","device":"60:a4:23:ff:fe:2c:8a:f7-01-0402","device_name":"RH3052: ZHATemperature","topic":"despatx","state":"temperature","output":"onupdate","outputAtStartup":true,"x":140,"y":140,"wires":[["46f5cb67.699984"],[]]},{"id":"bb6f0009.6f79b","type":"deconz-input","z":"335d5ee8.224672","name":"Sensor Temp Joan","server":"ea12a4.b85e1d6","device":"60:a4:23:ff:fe:2c:8b:b3-01-0402","device_name":"RH3052 : ZHATemperature","topic":"joan","state":"temperature","output":"onupdate","outputAtStartup":true,"x":150,"y":200,"wires":[["46f5cb67.699984"],[]]},{"id":"beb33a87.744328","type":"deconz-input","z":"335d5ee8.224672","name":"Sensor Temp Matrimoni","server":"ea12a4.b85e1d6","device":"60:a4:23:ff:fe:2c:8b:4e-01-0402","device_name":"RH3052: ZHATemperature","topic":"matrimoni","state":"temperature","output":"onupdate","outputAtStartup":true,"x":140,"y":80,"wires":[["46f5cb67.699984"],[]]},{"id":"5aa6b967.1993a8","type":"function","z":"335d5ee8.224672","name":"Converteix a decimal","func":"msg.payload = msg.payload/100;\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":640,"y":160,"wires":[["31a7cddc.7739d2"]]},{"id":"691ca4fb.7f7cbc","type":"deconz-input","z":"335d5ee8.224672","name":"Sensor Temp  Bany","server":"ea12a4.b85e1d6","device":"60:a4:23:ff:fe:2c:8a:f5-01-0402","device_name":"RH3052: ZHATemperature","topic":"bany","state":"temperature","output":"onupdate","outputAtStartup":true,"x":150,"y":260,"wires":[["46f5cb67.699984"],[]]},{"id":"31a7cddc.7739d2","type":"ui_chart","z":"335d5ee8.224672","name":"","group":"c02bbdd0.2d8d","order":1,"width":"8","height":"6","label":"Temperatura Casa","chartType":"line","legend":"true","xformat":"HH:mm:ss","interpolate":"linear","nodata":"","dot":true,"ymin":"","ymax":"","removeOlder":"2","removeOlderPoints":"","removeOlderUnit":"3600","cutout":0,"useOneColor":false,"useUTC":false,"colors":["#1f77b4","#aec7e8","#ff7f0e","#2ca02c","#98df8a","#d62728","#ff9896","#9467bd","#c5b0d5"],"useOldStyle":false,"outputs":1,"x":650,"y":240,"wires":[[]]},{"id":"46f5cb67.699984","type":"delay","z":"335d5ee8.224672","name":"Aliniar medicions","pauseType":"timed","timeout":"30","timeoutUnits":"seconds","rate":"1","nbRateUnits":"10","rateUnits":"minute","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":true,"x":410,"y":160,"wires":[["5aa6b967.1993a8"]]},{"id":"ea12a4.b85e1d6","type":"deconz-server","name":"DeConz_Server","ip":"192.168.2.3","port":"80","apikey":"B369C020B3","ws_port":"443","secure":false,"polling":"15"},{"id":"c02bbdd0.2d8d","type":"ui_group","name":"Temperatura i Humitat","tab":"98234e0e.89595","order":3,"disp":true,"width":"8","collapse":false},{"id":"98234e0e.89595","type":"ui_tab","name":"Temperatura i Humitat","icon":"fa-thermometer-full","order":3,"disabled":false,"hidden":false}]
`
This example is from another flow, not a frigate one but is a good example to understand the rate limit node.

But be careful because it will "drop" all the alerts over the rate limit ( more than 1 per 2 second).
